lib/topics_filter.rb

```diff
@@ -101,6 +113,20 @@ def filter_status(status:, category_id: nil)
 
   def extract_and_validate_value_for(filter, values)
     case filter
+    when "activity-before", "activity-after", "created-before", "created-after",
+         "latest-post-before", "latest-post-after"
+      value = values.last
+
+      if value =~ /^(?<year>[12][0-9]{3})-(?<month>0?[1-9]|1[0-2])-(?<day>0?[1-9]|[12]\d|3[01])$/
+        year = $~[:year].to_i
+        month = $~[:month].to_i
+        day = $~[:day].to_i
```

Comment:
Personally I'm not a huge fan of ruby's global variables like this one because they look weird and not very googleable for someone who is learning ruby. Can we use `String#match` here?

Reply:
Sure thing I've updated the code to use `String#match` instead.